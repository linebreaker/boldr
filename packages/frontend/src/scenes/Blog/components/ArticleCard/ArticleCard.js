/* @flow */
import React from 'react';
import { connect } from 'react-redux';
import { format } from 'date-fns';
import classnames from 'classnames';
import { Card, CardTitle, CardText, CardActions, Media } from '@boldr/ui';
import Button from '@boldr/ui/Button';
import { StyleClasses } from '../../../../theme/styleClasses';
import { selectArticle } from '../../state/actions';

import TagBlock from '../TagBlock';

const BASE_ELEMENT = StyleClasses.ARTICLE_CARD;
type Props = {
  className: string,
  article: Article,
  dispatch: () => void,
};

export const ArticleCard = (props: Props) => {
  const formattedDate = format(props.article.createdAt, 'MM/DD/YYYY');
  const classes = classnames(BASE_ELEMENT, props.className);

  function transitionPost() {
    const { article } = props;
    props.dispatch(selectArticle(article));
  }
  const { title, image, slug, tags, excerpt } = props.article;
  return (
    <div className={classes}>
      <Card style={{ maxWidth: 600 }} className="md-block-centered">
        <Media>
          <img src={image} alt={`${title} feature image`} role="presentation" />
        </Media>
        <CardTitle title={title} subtitle={formattedDate} />
        <CardActions expander>
          <Button kind="primary" onClick={transitionPost} href={`/blog/${slug}`} outline>
            Read More
          </Button>
        </CardActions>
        <CardText expandable>
          {excerpt}
        </CardText>
        <TagBlock tags={tags} />
      </Card>
    </div>
  );
};

export default connect(state => state, null, null, { pure: true })(ArticleCard);
