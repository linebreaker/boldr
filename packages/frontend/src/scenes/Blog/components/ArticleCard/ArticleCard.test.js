import React from 'react';
import { shallow } from 'enzyme';
import { shallowToJson } from 'enzyme-to-json';
import { ArticleCard } from './ArticleCard';

const ArticleData = {
  featureImage: 'http://boldr.io/images/logo.png',
  title: 'Test Post',
  excerpt: 'An excerpt',
  slug: 'test-post',
  tags: [{ name: 'abc', id: 1 }],
};

describe('<ArticleCard />', () => {
  it('renders <ArticleCard /> without breaking', () => {
    const wrapper = shallow(<ArticleCard article={ArticleData} />);
    expect(wrapper.find('div').length).toBe(1);
  });

  it('accepts props and renders them.', () => {
    const wrapper = shallow(<ArticleCard article={ArticleData} />);
    expect(wrapper.instance().props.article.featureImage).toBe('http://boldr.io/images/logo.png');
    expect(wrapper.instance().props.article.title).toBe('Test Post');
    expect(wrapper.instance().props.article.excerpt).toBe('An excerpt');
    expect(wrapper.instance().props.article.slug).toBe('test-post');
  });
  it('renders snapshot', () => {
    const wrapper = shallow(<ArticleCard article={ArticleData} />);
    expect(shallowToJson(wrapper)).toMatchSnapshot();
  });
});
