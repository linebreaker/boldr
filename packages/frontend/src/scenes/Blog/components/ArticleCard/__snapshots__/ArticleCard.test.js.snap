// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ArticleCard /> renders snapshot 1`] = `
<div
  className="boldrui-article-card"
>
  <Card
    animate={true}
    className="md-block-centered"
    expanderIconChildren="keyboard_arrow_down"
    expanderIconClassName="material-icons"
    expanderTooltipPosition="left"
    style={
      Object {
        "maxWidth": 600,
      }
    }
  >
    <Media
      aspectRatio="16-9"
      component="section"
      forceAspect={true}
    >
      <img
        alt="Test Post feature image"
        role="presentation"
      />
    </Media>
    <CardTitle
      subtitle="Invalid Date"
      title="Test Post"
    />
    <CardActions
      expander={true}
    >
      <Button
        block={false}
        bordered={true}
        className=""
        disabled={false}
        href="/blog/test-post"
        htmlType="button"
        kind="primary"
        loading={false}
        onClick={[Function]}
        outline={true}
        prefix="boldrui"
        size="medium"
      >
        Read More
      </Button>
    </CardActions>
    <CardText
      component="section"
      expandable={true}
    >
      An excerpt
    </CardText>
    <TagBlock
      tags={
        Array [
          Object {
            "id": 1,
            "name": "abc",
          },
        ]
      }
    />
  </Card>
</div>
`;
